javaharness/java/arcs/android/impl/ArcsLocal.java

```diff
@@ -1,97 +0,0 @@
-package arcs.android.impl;
-
-import javax.inject.Inject;
-
-import arcs.api.ArcData;
-import arcs.api.Arcs;
-import arcs.api.ArcsEnvironment;
-import arcs.api.PECInnerPort;
-import arcs.api.PecPortManager;
-import arcs.api.PortableJson;
-import arcs.api.PortableJsonParser;
-import arcs.api.UiBroker;
-
-// This class implements Arcs API for callers within the same Android service
-// that hosts the Arcs Runtime.
-public class ArcsLocal implements Arcs {
```

Comment:
This class was useful for cases when you wanted to run an Arc from within the Arcs service (this is how the "ingestion-arc" would be started, it just wasn't part of the prototype, but it is part of the design). 
I think it's nice if you could use the same interface to run an Arc from any service - Arcs or other Android service, this is why I created the Arcs interface and ArcsLocal & ArcsAndroid implementations (the names might be changed, but is there a reason to get rid of ArcsLocal implementation as a concept?)

Reply:
IMO, having an Arcs local API exposed outside of this packages breaks the molecularity. There shouldn't be any Arc run inside ArcsService. If we connect Arc through the AIDL interface, an Arc's crash (like a crash in an ingestion particle) won't cause the crash of ArcsService, vice-versa. This is a pretty standard approach in Android. Android system services (like Notification) are bound to the system process but not inside the system process so they can be dynamically replaced and their crashes won't cause system crash.